### ISSUE TITLE
Cursor at the end of the Text 

### ISSUE THREAD

**Issue by arpu** [protectwise/troika#78]
Hey,

working on a basic Cursor blinking at the End of the Text, any Idea what i should do if the maxWidth wrap the Text? 
( using totalBlockSize ) 
( attached small showcast Video) 

![Peek 2020-09-20 10-06](https://user-images.githubusercontent.com/14173/93706725-5c5c3880-fb29-11ea-94a3-d48e9437a7e8.gif)

**Comment by lojjic** [comment]
@arpu You should use the utility functions for caret and selection positioning described here: https://github.com/protectwise/troika/blob/master/packages/troika-three-text/README.md#carets-and-selection-ranges

Something like: add a listener for the `synccomplete` event, and in its handler call `getSelectionRects` passing the string length for both start and end. That should give you back a zero-width rectangle for placing the caret.

**Comment by arpu** [comment]
wau, thx for your fast response! this works fantastic!

**Comment by arpu** [comment]
@lojjic  not sure if this is a bug or a problem on my side, but the getSelectionRects is not updated for a space in the first Line 


https://user-images.githubusercontent.com/14173/108220345-ed662880-7136-11eb-9516-6108e941bc80.mp4

**Comment by arpu** [comment]
all works fine in the second line  i get the  right position with 
```
 const catBox = getSelectionRects(e.target._textRenderInfo, 0 , this.data.value.length )
  right = catBox[catBox.length-1].right

```

**Comment by lojjic** [comment]
Hmm, it looks like getSelectionRects clips the rects to the overall bounds of the rendered glyphs, which may explain what you're seeing. I'll have to give some thought to whether that's the behavior we want or not (I think probably not, but need to make sure.)

In the meantime, you could maybe work around it by accessing textRenderInfo.caretPositions directly?

**Comment by lojjic** [comment]
Ah I see why I included that clipping, it's to avoid ugly things like this where the selection pokes out the right edge:

![image](https://user-images.githubusercontent.com/138124/108241934-99911a80-7109-11eb-8165-e2360ff99eec.png)

But I think it's the wrong approach for handling that, it needs to be smarter. I'll work on it.

**Comment by lojjic** [comment]
@arpu I've just released a 0.39.1 patch version which I think should fix your issue. Please let me know whether it works.
